src/mgr/ServiceMap.h

```diff
@@ -12,17 +12,20 @@
 #include "include/buffer.h"
 #include "msg/msg_types.h"
 
+#include <boost/optional.hpp>
+
 namespace ceph {
   class Formatter;
 }
 
 struct ServiceMap {
   struct Daemon {
-    uint64_t gid = 0;
+    boost::optional<uint64_t> gid;
```

Comment:
this change is not backward compatible. it breaks `readable.sh`.

```
104: **** failed to decode /var/ssd/ceph/ceph-object-corpus/archive/15.0.0-539-g191ab33faf/objects/ServiceMap::Service/08800c908429da57f2b09b5cc8463080 ****
104: error: buffer::malformed_input: void entity_addr_t::decode(ceph::buffer::v14_2_0::list::const_iterator&) no longer understand old encoding version 1 < 28
104: error: buffer::malformed_input: void entity_addr_t::decode(ceph::buffer::v14_2_0::list::const_iterator&) no longer understand old encoding version 1 < 28
104: **** failed to decode /var/ssd/ceph/ceph-object-corpus/archive/15.0.0-539-g191ab33faf/objects/ServiceMap/074825661e8e8a3fbcb1705c109ad5db ****
104: error: buffer::malformed_input: void entity_addr_t::decode(ceph::buffer::v14_2_0::list::const_iterator&) no longer understand old encoding version 1 < 28
104: error: buffer::malformed_input: void entity_addr_t::decode(ceph::buffer::v14_2_0::list::const_iterator&) no longer understand old encoding version 1 < 28
104: error: buffer::malformed_input: void entity_addr_t::decode(ceph::buffer::v14_2_0::list::const_iterator&) no longer understand old encoding version 1 < 28
```

@vshankar @batrick  https://tracker.ceph.com/issues/41424

Reply:
Thanks for catching that @tchaikov